### ISSUE TITLE
[DependencyInjection]: Extra overloads for registering listeners

### ISSUE THREAD

**Issue by BraedonWooding** [quartznet/quartznet#1852]
**Is your feature request related to a problem? Please describe.**
We use a more general listener that sits over all listener interfaces to generalize out events, currently due to how C# dependency injection works it's creating duplicate singleton instances

**Describe the solution you'd like**
Just adding the extra overloads.

New overloads;
```csharp
        // existing
        void AddSchedulerListener<T>() where T : class, ISchedulerListener;
        void AddSchedulerListener<T>(T implementationInstance) where T : class, ISchedulerListener;
        void AddSchedulerListener<T>(Func<IServiceProvider, T> implementationFactory) where T : class, ISchedulerListener;

        // existing
        void AddJobListener<T>(params IMatcher<JobKey>[] matchers) where T : class, IJobListener;
        void AddJobListener<T>(T implementationInstance, params IMatcher<JobKey>[] matchers) where T : class, IJobListener;
        void AddJobListener<T>(Func<IServiceProvider, T> implementationFactory, params IMatcher<JobKey>[] matchers) where T : class, IJobListener;

        // existing
        void AddTriggerListener<T>(params IMatcher<TriggerKey>[] matchers) where T : class, ITriggerListener;
        void AddTriggerListener<T>(T implementationInstance, params IMatcher<TriggerKey>[] matchers) where T : class, ITriggerListener;
        void AddTriggerListener<T>(Func<IServiceProvider, T> implementationFactory, params IMatcher<TriggerKey>[] matchers) where T : class, ITriggerListener;
```

**Describe alternatives you've considered**
Accepting the duplicates, this is *fine* but would be much better to have them.

**Additional context**
I will create a PR with these changes (and link it here once done) just created an issue since that seems to be how you track changes.

**Comment by BraedonWooding** [comment]
Resolved by PR #1853

**Comment by lahma** [comment]
Now also synced to main.
